trpc-case-study.md

```diff
@@ -0,0 +1,275 @@
+Since many of tRPC's features benefit from using a singular repository, it might not be the best option if the development team has a need or requirement for heavily silo'd frontend and backend.
+
+#### Scalability
+- In larger scale projects, in a larger team, its often beneficial to come up with a clear set of rules and a schema that everyone can work from. tRPC does not offer this.
+- If your app is getting increasingly more complex and require more nuanced requirements, tRPC might not be the best option.
+
+
+## Comparison
+
+[Web Dev Simplified](https://www.youtube.com/watch?v=lxnPMB0Jc7E) asks recently in a video if tRPC spells the end to ``REST/graphQL``. While his title is almost certainly to generate views and clicks, the video does highlight some of the unique benefits of tRPC as well as its simplicity of use. When possible, people tend to gravitate towards the option that provides them the simplest solution. tRPC often do so in an intelligent and seamless manner. 
+
+However, there are dilemmas and industry requirements that ``tRPC`` cannot properly solve, and in such cases there might be better options out there. One such popular option is ``graphQL``.
+
+### GraphQL
+In short, ``graphQL`` is a query language for APIs and a runtime for fulfilling those queries with your existing data. While ``REST`` APIs will always return the entire object the client is requesting, ``GraphQL`` allows you to query to your API and return only the data you're interested in. ``With GraphQL`` the client controls the data returned, not the server. 
+
+``GraphQL`` queries can access several properties at once by following references between them, making it possible to fetch all the data you need in a single request. Significantly reducing the bandwidth footprint. 
+
+Similar to ``tRPC``, ``GraphQL`` offers typesafety. ``GraphQL`` is organized in terms of types and fields, instead of endpoints. 
+
+#### Scalability and maintainance
+With ``GraphQL`` you can easily add new fields and types to the API without impacting or breaking existing queries. As a result the server code base becomes easier to maintain and allows for API growth with less hassle and complication.
+
+GraphQL is being used by industry juggernaughts such as ``Facebook``, ``Github``, ``Airbnb``, ``Twitter`` and many others.
+
+### Which is better? tRPC or GraphQL?
+``tRPC`` is still very much in its infancy, and while many of its features are excellent it cannnot match ``GraphQL's`` scalability and network efficiency. Nor does it need or try to. In my view, they can coexist not as competing technologies, but technologies that both offer their strength and weaknesses depending on use cases.
+
+*Use GraphQL if:*
+- Its a large scale project, with the need of well defined schemas, maintenance and future development requirements.
+- if your team is either heavily silo'd or use multiple programming languages.
+
+*Use tRPC if:*
+- You are looking to learn to create your own API's.
```

Comment:
```suggestion
- You are looking to learn to create your own APIs.
```
In this case `API's` could mean:
- Belonging to API (the API's test methods are passing)
- API is ... (the API's broken again)

But not:
- More than one API (the APIs are working correctly)